### PR TITLE
Initial support for UDP audio transport

### PR DESCRIPTION
**Pull request by firealarmss**
Very ugly implementation. Clean up coming in the future. Only sends audio and will not receive for now.

### CODE REVIEW

File: MBEVocoder/MBEVocoder.vcxproj [hunk 1]
```diff
@@ -29,26 +29,26 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'" Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v143</PlatformToolset>
```
Comment (gatekeep) [inline MBEVocoder/MBEVocoder.vcxproj:32]: Why are we changing the platform toolset?

File: MBEVocoderManaged/MBEVocoderManaged.vcxproj [hunk 2]
```diff
@@ -21,7 +21,7 @@
   <PropertyGroup Label="Globals">
     <VCProjectVersion>16.0</VCProjectVersion>
     <ProjectGuid>{58076EA9-5A8E-4F1F-BD53-737FB1F7A705}</ProjectGuid>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
```
Comment (gatekeep) [inline MBEVocoderManaged/MBEVocoderManaged.vcxproj:24]: Why are we changing the target framework version?

File: dvmbridge/FNE/FneBase.cs [hunk 3]
```diff
@@ -413,6 +413,7 @@ public PeerDetails()
         {
             /* stub */
         }
+
```
Comment (gatekeep) [inline dvmbridge/FNE/FneBase.cs:416]: [Code Style] Don't have trailing empty lines.

File: dvmbridge/FneSystemBase.P25.cs [hunk 4]
```diff
@@ -23,13 +23,15 @@
...
 using System.Threading.Tasks;
+using System.Net.Sockets;
 
 using Serilog;
 
 using dvmbridge.FNE;
 using dvmbridge.FNE.P25;
 
 using vocoder;
+using System.Net;
```
Comment (gatekeep) [inline dvmbridge/FneSystemBase.P25.cs:34]: Move this using to the top with the other C# system using (just to keep organization in-place for usings).

File: dvmbridge/FneSystemBase.P25.cs [hunk 5]
```diff
@@ -38,6 +40,7 @@ namespace dvmbridge
     /// </summary>
     public abstract partial class FneSystemBase
     {
+
```
Comment (gatekeep) [inline dvmbridge/FneSystemBase.P25.cs:43]: [Code Style] Shouldn't have empty new lines like this.

File: dvmbridge/FneSystemBase.P25.cs [hunk 6]
```diff
@@ -678,63 +682,95 @@ private void P25EncodeAudioFrame(byte[] pcm)
...
+                            Console.WriteLine(Program.Configuration.UdpSendAddress);
+                            udpClient.Send(audioData, audioData.Length, new IPEndPoint(destinationIP, Program.Configuration.UdpSendPort));
+                        }
+                    }
                 }
             }
-        }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Exception: " + ex.Message);
```
Comment (gatekeep) [inline dvmbridge/FneSystemBase.P25.cs:766]: Don't use Console.WriteLine; in this instance for handling and displaying an exception message use the Serilog logging system to log an error.

File: dvmbridge/FneSystemBase.P25.cs [hunk 7]
```diff
@@ -764,6 +800,7 @@ protected virtual void P25DataReceived(object sender, P25DataReceivedEvent e)
                 // ensure destination ID matches
                 if (e.DstId != Program.Configuration.DestinationId)
                     return;
+                   // Console.WriteLine("DST ID IGNORED");
```
Comment (gatekeep) [inline dvmbridge/FneSystemBase.P25.cs:803]: This seems like developmental messaging, perhaps this should be removed?

File: dvmbridge/FneSystemBase.cs [hunk 8]
```diff
@@ -34,6 +34,8 @@
 
 using NAudio.Wave;
 using NAudio.Wave.SampleProviders;
+using System.Net.Sockets;
+using System.Net;
```
Comment (gatekeep) [inline dvmbridge/FneSystemBase.cs:38]: Organize these System usings with the rest of the System usings.

File: dvmbridge/FneSystemBase.cs [hunk 9]
```diff
@@ -449,7 +451,7 @@ private void WaveIn_DataAvailable(object sender, WaveInEventArgs e)
                 float[] temp = new float[meterInternalBuffer.BufferedBytes];
                 this.meterProvider.Read(temp, 0, temp.Length);
 
-                if (audioDetect && !callInProgress)
+                if (audioDetect)
```
Comment (gatekeep) [inline dvmbridge/FneSystemBase.cs:454]: Why are we violating this? This logic check is to ensure the system will not simultaneously receive and transmit on the talkgroup.

File: dvmbridge/config.example.yml [hunk 10]
```diff
@@ -70,10 +71,16 @@ location: Anywhere, USA
 
 # Source "Radio ID" for transmitted audio frames
 sourceId: 1234567
-# Flag indicating the source "Radio ID" will be overridden from the detected
-# MDC1200 pre- PTT ID.
-overrideSourceIdFromMDC: false
```
Comment (gatekeep) [inline dvmbridge/config.example.yml:75]: Why are we removing this option?

**Review by gatekeep — COMMENTED** [review]
See review comments.

**Comment by firealarmss** [comment]
Accidental close. Will start to fix issues.

**Comment by faultywarrior** [comment]
All comments should now be addressed and resolved.

**Comment by gatekeep** [comment]
> lgtm but thats up to @gatekeep
> 
> Idk why it let me add a review

Anyone can review it -- but for merging it requires a review from someone with write access (i.e. a DVMProject member).

@faultywarrior @firealarmss please resolve the merge conflict problem when you can. I will review this ASAP after that.

**Review by gatekeep — APPROVED** [review]
This looks okay. Some minor ordering cleanup is needed (just to make my OCD feel better), but I can do that after the merge.

**Pull request merged**